src/transport/GroupSession.h

```diff
@@ -71,69 +99,65 @@ class GroupSession : public Session
     const NodeId mSourceNodeId;
 };
 
-/*
- * @brief
- *   An table which manages GroupSessions
- */
-template <size_t kMaxSessionCount>
-class GroupSessionTable
+class OutgoingGroupSession : public Session
```

Comment:
We don't have this distinction between outgoing and incoming sessions for other types of sessions, so why this for Groups?

Materially speaking, the only practical difference I can see is the presence of the `GetSubjectDescriptor()` method. I don't think that alone justifies the creation of two classes here IMHO, especially when the other types don't have that distinction..

Reply:
There are fundamental difference between group and secure session.

For secure session, no matter for handling incoming or sending outgoing packets, the session the same, it is a bidirectional session. The peer in the session has same meaning for both of them.

On the contract, the group session is not bidirectional, the group session can only send, it can't respond, which renders the asymmetric structure for sending and receiving. For example, the peer means different thing for sending and receiving side. For sending, peer means destination group. for receiving, peer means origination of the packet, There are not the same thing, even have different type.

Materially speaking, their member fields are not same. the node id for sending can be omitted, because we can get it from the fabric, once we gain the access of `FabricInfo` within `SessionManager`. I'm going to do it in a separate PR.